### PR TITLE
Subforum: Add members list

### PR DESCRIPTION
**Pull request by Will-Howard**
In this PR:
 - Add members list to sidebar
 - Also make it accessible as a modal when the sidebar is not visible
 - Replace the Subscribe/notifications section in the header with:
   - A Join button if they are not a member
   - A notifications icon which pops out a menu if they are a member (currently this menu doesn't show the right settings

My reasoning behind the Join/notifications UI:
 - The join/leave button is always accessible via the members list, so it's less important to always have it in the header
 - We do however want to nudge users towards joining the subforum
 - Space was getting a bit tight in the header with both buttons
 - The only downside of this are:
   - For non-members there are two join buttons in close proximity, which looks kind of bad
   - Users can't change their notification settings without joining the subforum. I think this is actually fine and is consistent with the current Subscribe UI in normal topic pages

Screenshot 2022-11-21 at 13 29 02
Screenshot 2022-11-21 at 13 29 10
Screenshot 2022-11-21 at 13 29 33
Screenshot 2022-11-21 at 13 29 46

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203410978598151) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/tagging/TagSubforumPage2.tsx [hunk 1]
```diff
@@ -392,15 +456,12 @@ const TagSubforumPage2 = ({classes}: {
...
-            tag={tag}
-            className={classes.notifyMeButton}
-            subscribeMessage="Subscribe"
-            unsubscribeMessage="Unsubscribe"
-            subscriptionType={subscriptionTypes.newTagPosts}
-          />
-        )}
+        {/* TODO change what appears in SubforumNotificationSettings list */}
+        {/* Join/Leave button always appears in members list, so only show join button here as an extra nudge if they are not a member */}
+        {!!currentUser && !editing && (isSubscribed ? <SubforumNotificationSettings tag={tag} currentUser={currentUser} className={classes.notificationSettings} /> : <SubforumSubscribeSection tag={tag} className={classes.joinBtn} />)}
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/TagSubforumPage2.tsx:461]: I ran into a bug where if I joined from the "Wiki" tab, then switch to the "Posts" tab, the "Members" list doesn't list me (until I refresh). It does work properly if I click that same button from the "Posts" tab. Prob not a blocker, feel free to record and address later.

File: packages/lesswrong/components/tagging/TagSubforumPage2.tsx [hunk 2]
```diff
@@ -148,13 +154,49 @@ export const styles = (theme: ThemeType): JssStyles => ({
...
+  joinBtn: {
+    alignItems: 'center !important',
+    padding: '4px 0 0 0 !important',
+    '& button': {
+      minHeight: 0,
+      fontSize: 14,
+      padding: 8
+    },
+    [theme.breakpoints.down('sm')]: {
+      padding: '2px 0 0 0 !important',
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/TagSubforumPage2.tsx:187]: Why all the `!important`s? :(

File: packages/lesswrong/components/tagging/SubforumSubscribeSection.tsx [hunk 3]
```diff
@@ -71,6 +71,7 @@ const SubforumSubscribeSection = ({
   
   const isSubscribed = currentUser?.profileTagIds?.includes(tag._id)
 
+  // TODO: Either reword or remove the tooltip here
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/SubforumSubscribeSection.tsx:74]: Is this a TODO for this PR?

**Review by s-cheng — APPROVED** [review]
This was my first time seeing subforum2 - looks nice so far! 😄 

I'll approve this since it's not being used by users yet, so you don't necessarily need to fix all the bugs before merging this PR. I noticed a couple that you might want to address:

First, there is a screen width range where the sidebars are gone but the mobile view isn't applicable, so for example there's no way to see the members list:
Screen Shot 2022-11-22 at 7 19 40 PM

Second, in dark mode the member bio text fadeout doesn't match the background color of the surrounding box:
Screen Shot 2022-11-22 at 7 23 09 PM

File: packages/lesswrong/components/tagging/TagSubforumPage2.tsx (hunk 1 above)
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/TagSubforumPage2.tsx:461]: Hmm I can't reproduce this. I'll make a note to look into it though

File: packages/lesswrong/components/tagging/TagSubforumPage2.tsx (hunk 2 above)
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/TagSubforumPage2.tsx:187]: The root class of `SubforumSubscribeSection` also sets these. We can refactor these away once the other subforum page is deprecated (that's the only other place `SubforumSubscribeSection` is used)

File: packages/lesswrong/components/tagging/SubforumSubscribeSection.tsx (hunk 3 above)
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/SubforumSubscribeSection.tsx:74]: Actually I think we've settled on continuing to require membership to write discussion comments so the tooltip is accurate again, I've removed the TODO

**Pull request merged**